lucet-wasi/src/wasi.rs

```diff
@@ -0,0 +1,597 @@
+#![allow(clippy::too_many_arguments)]
+
+pub use lucet_runtime::{self, vmctx::lucet_vmctx};
+pub use wasi_common::{wasi, wasi32, WasiCtx};
+
+use lucet_runtime::lucet_hostcall_terminate;
+use std::mem;
+use std::rc::Rc;
+use wasi_common::hostcalls::*;
+
+lucet_runtime::lucet_hostcalls! {
+
+#[no_mangle]
+pub unsafe extern "C" fn __wasi_proc_exit(
+    &mut _lucet_vmctx,
+    rval: wasi::__wasi_exitcode_t,
+) -> ! {
+    export_wasi_funcs();
+    lucet_hostcall_terminate!(rval);
+}
```

Comment:
I deferred doing so because 1. we never actually inspect the value of exitcode anywhere except to see whether its nonzero, and 2. there are in-progress renovations to witx wasi-common that will expose these enum types as Rust enums (and reject values that are outside the valid enum cases). I didn't think it was worth the trouble to update everything to some homegrown version if we're going to replace it soon anyway.

Reply:
Cool, if there's going to be a nicer replacement soon, let's just wait for that.